pennylane/grouping/group_observables.py

```diff
@@ -203,9 +204,9 @@ def group_observables(observables, coefficients=None, grouping_type="qwc", metho
 
            * list[list[Observable]]: A list of the obtained groupings. Each grouping
              is itself a list of Pauli word ``Observable`` instances.
-           * list[list[float]]: A list of coefficient groupings. Each coefficient
-             grouping is itself a list of the grouping's corresponding coefficients. This is only
-             output if coefficients are specified.
+           * list[tensor_like]: A list of coefficient groupings. Each coefficient
+             grouping is itself a tensor of the grouping's corresponding coefficients. This is only
```

Comment:
At the moment this function will return a list of numpy tensors even if the input coefficients were a list. Is there any problem with this?

Reply:
I'm not sure if there is any problem with this, but I would expect it to return a tensor of the same type that I put in, i.e., one that matches the interface.

Reply:
I added an if statement that converts things to lists or tuples if needed!